### PR TITLE
[WIP] Update documentation and codebase with new string representations of X, Y, Z and Identity

### PR DESCRIPTION
**Pull request by Qottmann**
Updating docs and codebase with `qml.X/Y/Z/I(*)` instead of `qml.PauliX/Y/Z(*)` and `qml.Identity(*)` following https://github.com/PennyLaneAI/pennylane/pull/5116 and https://github.com/PennyLaneAI/pennylane/pull/5138/

work in progress...

Are changes of this order of magnitude reasonable?
Ignore failing tests for now.
Also, I must have messed up something when creating the branch because of these older, unrelated commits though the diff seems fine.

ToDo

- [ ] Improve https://docs.pennylane.ai/en/stable/introduction/operations.html#qubit-operators
- [ ] Update outputs in docs with PauliX(wires=*)
- [ ] Update outputs in docs with SProds and other composites (i.e. with extra spaces and removing unnecessary brackets)
- [ ] All tests pass
- [ ] changelog

### CODE REVIEW

File: doc/releases/changelog-0.11.0.md [hunk 1]
```diff
@@ -102,7 +102,7 @@
   def subroutine():
       qml.Hadamard(wires=[0])
       qml.CNOT(wires=[0, 1])
-      qml.PauliX(wires=[1])
+      qml.X(1)
```
Comment (trbromley) [inline doc/releases/changelog-0.11.0.md:105]: We can keep all the historic changelog entries the same, except for the dev one.

File: doc/code/qml_fermi.rst [hunk 2]
```diff
@@ -51,18 +51,18 @@ The fermionic operators can be mapped to the qubit basis by using the
 Fermi sentences.
 
 >>> qml.jordan_wigner(qml.FermiA(1))
-(0.5*(PauliZ(wires=[0]) @ PauliX(wires=[1])))
-+ (0.5j*(PauliZ(wires=[0]) @ PauliY(wires=[1])))
+(0.5*(Z(0) @ X(1)))
++ (0.5j*(Z(0) @ Y(1)))
```
Comment (trbromley) [inline doc/code/qml_fermi.rst:55]: With other updates to the repr, is this what it'll now look like?

Comment (Qottmann) [reply doc/code/qml_fermi.rst:55]: Ah yes good point, will have to update those too

**Comment by Qottmann** [comment]
Easier to close this and start new in https://github.com/PennyLaneAI/pennylane/pull/5153

**Pull request closed without merging**